===== COMPLETE LIST OF CHANGED FILES (3 files) =====
# AUTHORITATIVE: this list is never truncated. The diff BODY below may be trimmed for
# very large PRs, so decide WHICH files/areas changed (e.g. web/src frontend) from THIS list.
web/src/components/ingestion/Custom.vue
web/src/views/HomeNoDataState.vue
web/src/views/LogStream.vue
===== END CHANGED FILES =====

diff --git a/web/src/components/ingestion/Custom.vue b/web/src/components/ingestion/Custom.vue
index 1ab3ca1c566..30c71e61b21 100644
--- a/web/src/components/ingestion/Custom.vue
+++ b/web/src/components/ingestion/Custom.vue
@@ -41,6 +41,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
         :label="t('ingestion.metricsLabel')"
       />
       <ORouteTab
+        v-if="store.state.zoConfig?.profiling_enabled"
         name="ingestProfiles"
         data-test="ingestion-custom-tab-ingestProfiles"
         :to="{
diff --git a/web/src/views/HomeNoDataState.vue b/web/src/views/HomeNoDataState.vue
index 8a12d761379..2ece086c46f 100644
--- a/web/src/views/HomeNoDataState.vue
+++ b/web/src/views/HomeNoDataState.vue
@@ -121,6 +121,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
           >{{ t("home.noDataState.shippers") }}</EmptyStateIngestionChip
         >
         <EmptyStateIngestionChip
+          v-if="store.state.zoConfig?.profiling_enabled"
           icon="bar-chart"
           data-test="home-no-data-profiles-btn"
           @click="go('profiles')"
diff --git a/web/src/views/LogStream.vue b/web/src/views/LogStream.vue
index e2d9ead9374..732dff9c16c 100644
--- a/web/src/views/LogStream.vue
+++ b/web/src/views/LogStream.vue
@@ -101,7 +101,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                   /></template>
                   {{ t("logStream.labelTraces") }}
                 </OToggleGroupItem>
-                <OToggleGroupItem value="profiles" size="sm">
+                <OToggleGroupItem
+                  v-if="store.state.zoConfig?.profiling_enabled"
+                  value="profiles"
+                  size="sm"
+                >
                   <template #icon-left
                     ><OIcon name="bar-chart" size="xs" class="shrink-0"
                   /></template>
